cmd/query/app/static_handler.go

```diff
@@ -94,11 +126,54 @@ func NewStaticAssetsHandler(staticAssetsRoot string, options StaticAssetsHandler
 		}
 		indexBytes = basePathPattern.ReplaceAll(indexBytes, []byte(fmt.Sprintf(basePathReplace, options.BasePath)))
 	}
-	return &StaticAssetsHandler{
-		options:   options,
-		indexHTML: indexBytes,
-		assetsFS:  assetsFS,
-	}, nil
+
+	return indexBytes, nil
+}
+
+func (sH *StaticAssetsHandler) watch() {
+	if sH.options.UIConfigPath == "" {
+		return
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		sH.options.Logger.Error("failed to create a new watcher for the UI config", zap.Error(err))
+		return
+	}
+	defer watcher.Close()
+
+	done := make(chan bool)
```

Comment:
I am not sure why this is needed, there's no  external signaling into this chan

Reply:
That was mainly to keep the function from returning, as it would then call `watcher.Close()`, but I figured I could just not close the watcher and it would have the same effect.